cupy/core/_routines_manipulation.pyx

```diff
@@ -706,11 +718,12 @@ cdef ndarray _concatenate_single_kernel(
         ptrs[i] = a.data.ptr
     x = core.array(ptrs)
 
-    ret = core.ndarray(shape, dtype=dtype)
+    if out is None:
+        out = core.ndarray(shape, dtype=dtype)
```

Comment:
I think `out` always has a value.
How about `assert out is not None` instead (at the beginning of the function)?